src/Stratis.Bitcoin.Features.Consensus.Tests/Rules/CommonRules/BlockHeaderPosContextualRuleTest.cs

```diff
@@ -0,0 +1,278 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using NBitcoin;
+using Stratis.Bitcoin.Features.Consensus.Rules.CommonRules;
+using Xunit;
+
+namespace Stratis.Bitcoin.Features.Consensus.Tests.Rules.CommonRules
+{
+    public class BlockHeaderPosContextualRuleTest : TestPosConsensusRulesUnitTestBase
+    {
+        public BlockHeaderPosContextualRuleTest() : base()
+        {
+            this.concurrentChain = GenerateChainWithHeight(5, this.network);
+            this.consensusRules = this.InitializeConsensusRules();
+        }
+
+        [Fact]
+        public async Task RunAsync_HeaderVersionBelowSeven_ThrowsBadVersionConsensusErrorAsync()
+        {
+            this.ruleContext.SkipValidation = false;
+            this.ruleContext.BlockValidationContext.ChainedBlock = this.concurrentChain.GetBlock(3);
+            this.ruleContext.BlockValidationContext.ChainedBlock.Header.Version = 6;
+
+            var exception = await Assert.ThrowsAsync<ConsensusErrorException>(() => this.consensusRules.RegisterRule<BlockHeaderPosContextualRule>().RunAsync(this.ruleContext));
```

Comment:
The reason for this is because in many cases I need to call Initialize on the rule and set the consensusrules and logger to the rule. The consensusrules class does all this for me which is a class I need an instance of anyway because it's used by the rule in many of the cases. I can rewrite it like you state no problem what do you think? I think the end result of rewriting will be better.

Reply:
@majikandy just let me know and I'll start doing the rest of the changes after you replied. I already have local commits to remove the base constructor calls and the skipvalidation.